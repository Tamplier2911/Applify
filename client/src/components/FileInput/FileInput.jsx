// import "./FileInput.scss";
import React from "react";
import PropTypes from "prop-types";

// JS Rendering CSS
import {
  FileInputContainer,
  FileInputLabel,
  FileInputSign,
  FileInputElement,
} from "./FileInputStyles";

const FileInput = ({ onInputChange, label, id, image, ...otherProps }) => {
  return (
    <FileInputContainer data-test="file-input">
      {label ? (
        <FileInputLabel data-test="file-input-label" htmlFor={id}>
          {label}
        </FileInputLabel>
      ) : null}
      {image ? (
        <FileInputSign image={image}>&#10004;</FileInputSign>
      ) : (
        <FileInputSign image={image}>&#10006;</FileInputSign>
      )}
      <FileInputElement
        data-test="file-input-element"
        onChange={onInputChange}
        id={id}
        {...otherProps}
      />
    </FileInputContainer>
  );
};

FileInput.propTypes = {
  onInputChange: PropTypes.func,
  label: PropTypes.string,
  id: PropTypes.string,
  image: PropTypes.number,
  otherProps: PropTypes.object,
};

export default FileInput;
