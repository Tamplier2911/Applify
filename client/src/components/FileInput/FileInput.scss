/*

.file-input {
  position: relative;

  &__label {
    cursor: pointer;
    display: inline-block;
    font-family: var(--font-title);
    font-size: 1.7rem;
    transition: color 0.3s;
    padding: 0.6rem 0rem;

    &:hover {
      color: var(--cl-primary);
    }
  }

  &__input {
    display: none;
  }
}

*/
