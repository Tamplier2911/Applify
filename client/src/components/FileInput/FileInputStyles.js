import styled from "styled-components";

export const FileInputContainer = styled.div`
  position: relative;
`;

export const FileInputLabel = styled.label`
  cursor: pointer;
  display: inline-block;
  font-family: var(--font-title);
  font-size: 1.7rem;
  transition: color 0.3s;
  padding: 0.6rem 0rem;

  &:hover {
    color: var(--cl-primary);
  }
`;

export const FileInputElement = styled.input`
  display: none;
`;
