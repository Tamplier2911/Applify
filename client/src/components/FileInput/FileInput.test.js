import React from "react";

// component
import FileInput from "./FileInput";

// utils
import { setShallow, findByTestAttr, checkProps } from "../../tests/testUtils";

describe("<FileInput />", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = setShallow(<FileInput />);
  });

  it("renders without an error", () => {
    const fileInput = findByTestAttr(wrapper, "file-input");
    expect(fileInput.length).toBe(1);
  });

  it("renders child - label if label property is passed in", () => {
    wrapper = setShallow(<FileInput label={"Labelicous"} />);
    const fileInputLabel = findByTestAttr(wrapper, "file-input-label");
    expect(fileInputLabel.length).toBe(1);
  });

  it("renders child - input element", () => {
    const fileInputElement = findByTestAttr(wrapper, "file-input-element");
    expect(fileInputElement.length).toBe(1);
  });

  it("recieves expected props", () => {
    const expectedProps = {
      onInputChange: () => {},
      label: "Labelous",
      id: "SDK2131",
      image: 1,
      otherProps: {},
    };
    const err = checkProps(FileInput, expectedProps, "props");
    expect(err).toBeUndefined();
  });

  it("", () => {
    const fileInput = findByTestAttr(wrapper, "file-input");
    expect(fileInput).toMatchSnapshot();
  });
});
