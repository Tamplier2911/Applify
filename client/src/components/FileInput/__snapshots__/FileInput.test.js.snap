// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FileInput />  1`] = `ShallowWrapper {}`;
